src/CoreFoundation/CFProxySupport.cs

```diff
@@ -602,19 +609,27 @@ public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (NSString prox
 
 			if (resultCallback == null)
 				throw new ArgumentNullException ("resultCallback");
-
-			if (clientContext == null)
-				throw new ArgumentNullException ("clientContext");
-
-			IntPtr source = CFNetworkExecuteProxyAutoConfigurationScript (proxyAutoConfigurationScript.Handle, targetURL.Handle, resultCallback, clientContext);
-			return (source == IntPtr.Zero) ? null : new CFRunLoopSource (source);
+			unsafe {
+				BlockLiteral block_handler = new BlockLiteral ();
+				BlockLiteral *block_ptr_handler = &block_handler;
+				block_handler.SetupBlockUnsafe (static_AutoConfigurationResult, resultCallback);
+		
+				try {
+					IntPtr clientPtr = Marshal.AllocHGlobal (Marshal.SizeOf (clientContext));
+					Marshal.StructureToPtr (clientContext, clientPtr, false);
+					IntPtr source = CFNetworkExecuteProxyAutoConfigurationScript (proxyAutoConfigurationScript.Handle, targetURL.Handle, (void*) block_ptr_handler, clientPtr);
+					return (source == IntPtr.Zero) ? null : new CFRunLoopSource (source);
+				} finally {
+					block_handler.CleanupBlock ();
+				}
+			}
 		}
 		
 		[DllImport (Constants.CFNetworkLibrary)]
-		extern static /* CFRunLoopSourceRef __nonnull */ IntPtr CFNetworkExecuteProxyAutoConfigurationURL (
+		extern unsafe static /* CFRunLoopSourceRef __nonnull */ IntPtr CFNetworkExecuteProxyAutoConfigurationURL (
 			/* CFURLRef __nonnull */ IntPtr proxyAutoConfigurationURL,
 			/* CFURLRef __nonnull */ IntPtr targetURL,
-			/* CFProxyAutoConfigurationResultCallback __nonnull */ IntPtr cb,
+			/* CFProxyAutoConfigurationResultCallback __nonnull */ void* cb,
 			/* CFStreamClientContext * __nonnull */ IntPtr clientContext);
 
 		public static CFRunLoopSource ExecuteProxyAutoConfigurationURL (NSUrl proxyAutoConfigurationURL, NSUrl targetURL, CFProxyAutoConfigurationResultCallback resultCallback, CFStreamClientContext clientContext)
```

Comment:
Casing: `ExecuteProxyAutoConfigurationUrl`

A signature like this would be much nicer:

```csharp
public delegate void AutoConfigurationResultCallback (NSArray proxyList, NSError error);
public static CFRunLoopSource ExecuteProxyAutoConfigurationUrl (NSUrl proxyAutoConfigurationUrl, NSUrl targetUrl, AutoConfigurationResultCallback callback)
{
    // ...
}
```